cloud_secrets/providers/vault_provider.py

```
@@ -0,0 +1,211 @@
+            jwt = Path(SERVICE_ACCOUNT_TOKEN_PATH).read_text()
+        except OSError as e:
+            raise ConfigurationError(
+                f"Vault role '{self._role}' needs a Kubernetes service account "
+                f"token at {SERVICE_ACCOUNT_TOKEN_PATH}: {e}"
+            )
+
+        try:
+            self.client.auth.kubernetes.login(
+                role=self._role, jwt=jwt, mount_point=self._auth_mount_point
+            )
+        except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)